astropy/units/tests/test_typing.py

```diff
@@ -0,0 +1,110 @@
+# coding: utf-8
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Test utilities for :mod:`astropy.units._typing`.
+
+These tests are skipped if :mod:`~typing` or ``typing_extensions```
+have type ``Annotated``.
+
+"""
+
+import pytest
+
+try:  # py3.9+ builtin
```

Comment:
This test file still needs clean up. I also think it is should either check always - since presumably we are mimicking the existing class and might as well know that that one behaves the same - or just forego testing altogether (i.e., rely on the other tests), i.e., just remove the whole file (the only reason to suggest this is that your set of tests is great!).

Reply:
I want to move `_typing` to extern. There doesn't seem to be any test files in there though...

Reply:
For now, I'm putting the tests in ``astropy/tests``.